perf(ui): improve performance on resizing

ipy opened the pull request:
解决了调整大小时卡顿的问题。

注释掉了一段代码和对应的测试，这段代码的意图应该在是在窗口获得焦点时，如果鼠标在窗口内就开始播放。不过：
* 代码有点问题并没有正常工作
* 开始播放是个比较危险的动作并不适合在窗口聚焦时不经过用户就自动开始
这个需要确认下 @huzhenghui37

ipy committed:
fix(multi-platform): fix file url processing for Windows

ipy committed:
fix(multi-platform): make it possible to choose file and resize window on linux

ipy committed:
perf(ui): improve performance on resizing

huzhenghui37 commented:
这个功能我知道，是梁以昌同学以前写的，当时是希望播放窗口没有焦点时点击播放窗口获得焦点，如果处于暂停播放状态，则直接开始播放，如果是正在播放，则只是获得焦点。没想到会影响性能。

huzhenghui37 commented:
果然快了很多，哈哈。

tomasen commented on src/renderer/components/LandingView.vue:
```diff
@@ -203,7 +205,7 @@ export default {
       }
     });
-    if (process.platform === 'win32') {
+    if (process.platform !== 'darwin') {
```
这里为什么换成这种写法？是确定Linux也不支持drag么？

ipy replied:
嗯，亲测不可用...

ipy commented:
我表达有误，这个功能跟性能没关系，是由于这个功能没通过测试，所以看了下。现在这个功能有点问题，并没有实现暂停时开始播放。而且个人觉得这个功能本身并不是特别合适，所以找你确认下。如果确定要做我就把这个 bug 修一下，不做的话就把注释掉的代码删掉。 @huzhenghui37

tomasen commented on src/renderer/components/LandingView/Playlist.vue:
```diff
@@ -153,21 +155,25 @@ export default {
       const browserWindow = remote.BrowserWindow;
       const focusedWindow = browserWindow.getFocusedWindow();
-      const VALID_EXTENSION = [];
+      const VALID_EXTENSION = ['3g2', '3gp', '3gp2', '3gpp', 'amv', 'asf', 'avi', 'bik', 'bin', 'crf', 'divx', 'drc', 'dv', 'dvr-ms', 'evo', 'f4v', 'flv', 'gvi', 'gxf', 'iso', 'm1v', 'm2v', 'm2t', 'm2ts', 'm4v', 'mkv', 'mov', 'mp2', 'mp2v', 'mp4', 'mp4v', 'mpe', 'mpeg', 'mpeg1', 'mpeg2', 'mpeg4', 'mpg', 'mpv2', 'mts', 'mtv', 'mxf', 'mxg', 'nsv', 'nuv', 'ogg', 'ogm', 'ogv', 'ogx', 'ps', 'rec', 'rm ...
```
有来源么？有些担心这里会不会不全

ipy replied:
Copied from VLC. 只是临时解决一下，原来用空数组在 Mac 上没问题，但是其他平台上无法选择文件，必须要传入一些扩展名。

tomasen replied:
了解了

tomasen commented on src/renderer/components/PlayingView.vue:
```diff
@@ -245,12 +249,28 @@ export default {
+  watch: {
+    isFocused(isFocused, prevIsFocused) {
+      if (isFocused && !prevIsFocused) {
+        // // TODO: move isPlaying to global state to prevent nested refs
```
这里的功能应该如eric所说，还是需要的。

ipy replied:
👌

tomasen replied:
确定不能用就删掉吧。也不要注释了

tomasen replied:
或者/* */ 注释中解释清楚为什么

tomasen commented on test/unit/specs/PlayingView.spec.js:
```diff
@@ -204,19 +208,20 @@ describe('PlayingView.vue.lyc', () => {
-    win.focusWindow();
-    expect(vue.$refs.VideoCanvasRef.$refs.videoCanvas.paused).equal(false);
+    // TODO: to be determined
+    // const win = vue.mainWindow;
```
如前

tomasen replied:
确定不能用就删掉吧。也不要注释了

huzhenghui37 commented:
注释掉吧，貌似也不妥。刚刚试了一下这个PR版本，缩放窗口确实快了不少。

ipy committed:
fix(ux): always play when click window to focus

The previous implementation detects the focus event and check if the mouse is in the window. It may make the video start playing when focus using keyboard shortcut. The current implementation detects the first mouse click and check if it is triggered right after being focused.

ipy committed:
Merge branch 'develop' of github.com:chiflix/splayerx into develop

ipy commented:
调整了下，原先 Cmd+Tab 聚焦的时如果鼠标在窗口内也会开始播放；现在必须通过鼠标点击聚焦才会播放。
